pkg/resource/aws/deserializer/s3_bucket_analytic_deserializer.go

```diff
@@ -23,8 +23,10 @@ func (s S3BucketAnalyticDeserializer) HandledType() resource.ResourceType {
 func (s S3BucketAnalyticDeserializer) Deserialize(rawAnalytic []cty.Value) ([]resource.Resource, error) {
 	var inventories []resource.Resource
 	for _, analytic := range rawAnalytic {
+		analytic := analytic
```

Comment:
What's the point of this assignment?

Reply:
https://medium.com/@Nidhi11594/variable-shadowing-in-golang-f500e8e58931

I just know this is about shadowing variable, engineering team discuss about it before to avoid error on this kind of loop. You will probably get deeper answer from wbeuil.

Reply:
Ok, I'm just not sure what is the purpose here since  `analytic` variable is declared the line before, so if there was another `analytic` var declared earlier in the code, the scope of that loop would just ignore it, isn't what we want ?

Reply:
Our linter scopelint were outputting errors for unpinned variables here: 
https://app.circleci.com/pipelines/github/cloudskiff/driftctl/991/workflows/80798d6d-ff0e-4aaf-afe1-e9ce1625f00d/jobs/1481

Good examples in its repo https://github.com/kyoh86/scopelint

Reply:
Alright got it, thanks :)